mobile/src/main/res/values/strings.xml

```diff
@@ -72,13 +72,13 @@
     <string name="error_network_not_available">Network is not available</string>
     <string name="error_http_connection_failed">Received an invalid response from openHAB. Please check if openHAB is working correctly (HTTP response code %d)</string>
```

Comment:
Any idea how to improve this string? I stumbled over it a couple of times during working with translations.
The second sentence doesn't add value, also I don't like the mentioning of "openHAB" without any specifier like "server".

Reply:
I can remove the second sentence and add "server" to openhab.

Reply:
Sounds good. We could also hold on to the idea of the second sentence but describe possible causes better:

> An invalid response was received from the openHAB server (HTTP response code %d). Please check *this or that*.

What could actually lead to this situation?

Reply:
That heavily relies on the returned http status code. There're special messages, iirc, for common status codes, like 500, 403, 401 and so on, where we most likely know the cause, or can point to somewhere, so the user can find out easily where the problem is. However, this message is used, when we do not have a separate message for the status code. That itself already means, that we most likely do not know the cause of the problem. Suggesting things here is pretty complicated and I would be against anything here, as we can not make sure, that our suggestion is even a good idea in all possible http status codes :(

Reply:
"HTTP error 42 occured, when connecting to you openHAB server"?

Reply:
I'd not move the error code into more focus. That's a technical detail the average user will not be able to understand....

Actually this is exactly why I started this discussion. Put yourself into the position of an average no-technical end user of the app. Upon reading the error message, what did you learn? Something seems to be off. The message doesn't offer more insight.

I agree with both of you that (1) it's not easy to interpret the error message and (2) suggest next steps.

To that end, how about a more detailed message?

> Connection unsuccessful. An unexpected response was received while trying to connect to the configured openHAB server. Please ensure a working network connection, check the accessibility and operation of the server or contact your sever administrator. (Received HTTP response code: %d)

Reply:
That one sounds good. Should other http error strings also be extended with the second sentence?

Reply:
As you wish. Where it makes sense, why not. ;)

Reply:
I'm not sure whether that 5 line message, even though it's way more helpful for the user, is well suited here: I simply think it's too long for the amount of time it's shown. For Snackbars we could solve it by using DURATION_INFINITE instead of DURATION_LONG for certain messages, but that isn't an option for toasts. I'm really not sure a nontechnical user fully understands such verbose messages shown in a toast...

Reply:
Well... doesn't sound like a message that should be shown in a toast, should it? Toast messages are for unimportant confirmations and stuff.
I'm not the android development expert here but toasts are imho seldom used for important stuff.

Reply:
That's right, but spawning a foreground dialog from a background service (e.g. voice recognition from widget) isn't good UI either.

Reply:
@maniac103 Is right, I forgot that these messages are displayed in toasts, when using the voice widget. I shortend the messages.